### ISSUE TITLE
No longer possible to prevent redirections?

### ISSUE THREAD

**Issue by janwerkhoven** [simplabs/ember-simple-auth#876]
Hi,

We upgraded our apps to Ember 2.3.0 and Simple Auth 10.1.

**None of our Ember apps require a page redirect after login / logout.**

In Simple Auth 0.8.0 we managed to prevent Simple Auth's redirections after validation / invalidation by adding these empty actions to the application route:

```
actions: {
    sessionAuthenticated: function() {},
    sessionInvalidated: function() {}
}
```

In Simple Auth 1.0.1 this is no longer possible? We also tried these:

```
actions: {
    sessionInvalidationSucceeded: function() {},
    sessionAuthenticationSucceeded: function() {}
}
```
Any ideas?

-----

PS: Much love your work. Been using Simple Auth since the start.

**Comment by marcoow** [comment]
These are no longer actions but plain methods now - see docs: http://ember-simple-auth.com/api/classes/ApplicationRouteMixin.html

**Comment by janwerkhoven** [comment]
Thanks.

Your documentation might be outdated?



```Javascript
// Ember: 2.3.0
// Ember Data: 2.3.3
// Ember Simple Auth: 1.0.0
// Ember CLI: 2.2.0-beta.6
// app/instance-initializers/session-events.js

Ember.Application.initializer({
  name:       'session-events',
  after:      'ember-simple-auth',
  initialize: function(container, application) {
    var applicationRoute = container.lookup('route:application');
    var session          = container.lookup('service:session');
    session.on('authenticationSucceeded', function() {
      applicationRoute.transitionTo('index');
    });
    session.on('invalidationSucceeded', function() {
      window.location.reload();
    });
  }
});

```
Results in

 ```Uncaught Error: Assertion Failed: An instance initializer cannot be registered without an initialize function```.

Fixing the wrappers:

```Javascript
import Ember from 'ember';
export default {
  // app/instance-initializers/session-events.js
  name: 'session-events',
  after: 'ember-simple-auth',
  initialize: function(container, application) {
    var applicationRoute = container.lookup('route:application');
    var session = container.lookup('service:session');
    ...
  }
};
```

Then here `container.lookup('route:application')` errors to `Uncaught ReferenceError: container is not defined(…)`. Might be new in Ember?

The instance initializer works and the events fire. However the transition is not prevented by clearing the the event functions, nor returning false. If no transition is given it uses the default. I had to add [**this hack**](http://beerlington.com/blog/2015/02/11/ember-simple-auth-stay-on-current-page-after-authentication/) to every login route (we have multiple, all pointing at different login destinations).

```Javascript
  // Hack to prevent Simple Auth from redirecting after AuthenticationSuccess
  beforeModel: function(transition) {
    if (!this.get('session.isAuthenticated')) {
      this.set('session.attemptedTransition', transition);
    }
  },
```

**Comment by marcoow** [comment]
@janwerkhoven updated docs will be released with ESA 1.0.1.

In general, when you're using the `ApplicationRouteMixin` you get redirect after login and reloading after logout out of the box. If you don't want these to happen automatically you just don't use the mixin at all.

**Comment by janwerkhoven** [comment]
All right, thanks for the heads up :+1:
